packages/@jsii/kernel/src/kernel.ts

```diff
@@ -159,10 +164,15 @@ export class Kernel {
       );
     }
 
+    // We do a `require.resolve` call, as otherwise, requiring with a directory will cause any `exports` from
+    // `package.json` to be ignored, preventing injection of a "lazy index" entry point.
+    const entryPoint = this.#require!.resolve(assmSpec.name, {
+      paths: [this.#installDir!],
+    });
```

Comment:
There is an issue with this approach. tl;dr it doesn't work if there are no exports defined.

Some details here: https://github.com/projen/projen/issues/2644